### PR TITLE
Add permissions

### PR DESCRIPTION
**Pull request by MrDiben**

### CODE REVIEW

File: django_socio_grpc/request_transformer/grpc_socio_proxy_context.py [hunk 1]
```diff
@@ -30,14 +43,20 @@ def get_query_params(self, grpc_request_metadata):
     def build_absolute_uri(self):
         return "NYI"
 
+    def grpc_action_to_http_method_name(self, grpc_action):
+        for k, v in self.METHOD_MAP.items():
+            if grpc_action.lower().startswith(k.lower()):
```
Comment (AMontagu) [inline django_socio_grpc/request_transformer/grpc_socio_proxy_context.py:48]: why a startswith here ?

File: django_socio_grpc/tests/test_permissions.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        ) as mock_has_object_permissions:
+            mock_has_object_permissions.return_value = True
+            dummy_service.check_object_permissions("fake_obj")
+            mock_has_object_permissions.assert_called_once_with(
+                dummy_service.context, dummy_service, "fake_obj"
+            )
+
+    @mock.patch("django_socio_grpc.services.Service.permission_denied")
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_object_permissions_not_ok(
```
Comment (AMontagu) [inline django_socio_grpc/tests/test_permissions.py:88]: can you check that FakeContext.abort has been correctly call too ?

File: django_socio_grpc/tests/test_permissions.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            mock_has_object_permissions.assert_called_once_with(
+                dummy_service.context, dummy_service, "fake_obj"
+            )
+            mock_permission_denied.assert_called_once_with(
+                message="fake message", code="fake code"
+            )
+
+    def test_get_permissions(self):
+        dummy_service = DummyService()
+        dummy_service.permission_classes = [FakePermission]
```
Comment (AMontagu) [inline django_socio_grpc/tests/test_permissions.py:110]: in the other test FakePermission is called and here it is not why ? (FakePermission() -> FakePermission)

File: django_socio_grpc/request_transformer/grpc_socio_proxy_context.py (hunk 1 above)
Comment (MrDiben) [reply django_socio_grpc/request_transformer/grpc_socio_proxy_context.py:48]: To match http method (GET, POST...) we need to call our service method (List..., Create...)

File: django_socio_grpc/tests/test_permissions.py (hunk 3 above)
Comment (MrDiben) [reply django_socio_grpc/tests/test_permissions.py:110]: self.get_permissions() return instance of permission objects and permission_classes attr is a list of permission classes

File: django_socio_grpc/tests/test_permissions.py (hunk 2 above)
Comment (MrDiben) [reply django_socio_grpc/tests/test_permissions.py:88]: I already have an unitary test for permission_denied method that assert context.abort is called.
In this method I mock permission_denied to make sure it is called with correct args.

File: django_socio_grpc/request_transformer/grpc_socio_proxy_context.py [hunk 4]
```diff
@@ -44,10 +44,10 @@ def build_absolute_uri(self):
         return "NYI"
 
     def grpc_action_to_http_method_name(self, grpc_action):
-        for k, v in self.METHOD_MAP.items():
-            if grpc_action.lower().startswith(k.lower()):
-                return v
-        return None
+        try:
+            return self.METHOD_MAP[grpc_action]
```
Comment (AMontagu) [inline django_socio_grpc/request_transformer/grpc_socio_proxy_context.py:48]: use a .get method with a default to None

**Review by AMontagu — APPROVED** [review]

**Pull request merged**